Source/WebCore/platform/graphics/gstreamer/mse/PlaybackPipeline.cpp

```diff
@@ -394,6 +386,10 @@ void PlaybackPipeline::flush(AtomicString trackId)
 
     if (!gst_base_src_new_seamless_segment(GST_BASE_SRC(appsrc), segment->start, segment->stop, segment->start)) {
         GST_WARNING("Failed to send seamless segment event for trackId=%s", trackId.string().utf8().data());
+    }
```

Comment:
Sending the flush-stop event after the segment one breaks the order of events expected in the GStreamer design. The expected order is flush-start, flush-stop, segment. This is documented here:
https://gstreamer.freedesktop.org/documentation/plugin-development/advanced/events.html?gi-language=c#flush-stop

Reply:
gst_base_src_new_seamless_segment doesn't send the event. it just configures a new seamless segment to be sent on streaming thread. according to gst doc it should be used with stream-lock held. But, since we're anyways stopping the streaming thread and flushing the video part of the pipeline, it is enough to make sure the seamless segment is configured before streaming is restarted. Hence it is sent before flush stop. 

The base src class will ensure events are sent in order.